lib/waistband/model.rb

```diff
@@ -0,0 +1,195 @@
+require 'digest/md5'
+require 'active_support/core_ext/object/blank'
+require 'active_model'
+
+module Waistband
+  class Model
+
+    extend ::ActiveModel::Callbacks
+
+    include ::ActiveModel::Validations
+    include ::ActiveModel::Validations::Callbacks
+
+
+    #################
+    # Class methods #
+    #################
+
+    class << self
+
+      attr_reader :type_name, :column_defaults
+
+      def index_name(name)
+        @index_name = name.to_s
+      end
+
+      def index_type(name)
+        @type_name = name.to_s
+      end
+
+      def index
+        raise "index_name not defined!" if @index_name.blank?
+        Waistband::Index.new(@index_name)
+      end
+
+      def defaults(defaults)
+        @column_defaults ||= {}
+        @column_defaults.merge!(defaults)
+      end
+
+      def find(id)
+        found_attributes = index.find(id)
+
+        unless found_attributes
+          raise ::Waistband::Errors::Model::NotFound.new("Couldn't find #{self} with 'id'=#{id}")
+        end
+
+        new(found_attributes.merge(id: id))
```

Comment:
I think you should also set `persisted: true`, somehow, because I think that right now if you find the record and do `new_record?` it will return true ?